pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -989,3 +945,43 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 func createUID(podName, namespace string) string {
 	return "kubernetes://" + podName + "." + namespace
 }
+
+// first phase of instantiating IstioEndpoint
+func (c *Controller) newIstioEndpoint(pod *v1.Pod) model.IstioEndpoint {
```

Comment:
this seems like an odd pattern to me. Normally when there are "builders" the code looks like `newEndpoint().SetPod(pod).Setaddress(address).Build()` or something. This way, people might call `newIstioEndpoint` or `completeIstioendpoint` without calling the other method. Looking at the code I was pretty confused what was going on initially.

Reply:
I would come up with better name, and if you have any suggestion, PLMK